pallets/proof_of_existence/src/benchmarking.rs

```diff
@@ -13,23 +13,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![cfg(feature = "runtime-benchmarks")]
+
 use super::*;
 
-use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use frame_benchmarking::v2::*;
 use frame_system::RawOrigin;
 use sp_core::{Get, H256};
+use sp_runtime::SaturatedConversion;
+
+#[benchmarks]
+mod benchmarks {
+    use super::*;
+
+    const OFFSET_FACTOR: u64 = 10000;
+    #[benchmark]
+    fn publish_attestation() {
+        // Setup code
+        // benchmark taking into account situations where attestations left to be cleaned in storage
+        // are bigger than one. shouldn't happen very often so we consider only 0.01% of the
+        // maximum supported ones.
+        let max_attestations = T::MaxStorageAttestations::get().saturated_into::<u64>();
+        let offset = max_attestations
+            .div_ceil(OFFSET_FACTOR)
+            .saturated_into::<u64>();
 
-benchmarks! {
-    publish_attestation {
-        // benchmark with double the minimum number of elements for publishing an attestation
         let mut hash: [u8; 32] = [0; 32];
-        for h in 0 .. T::MinProofsForPublishing::get() * 2 {
-            hash[0] += 1;
-            Values::<T>::insert(0, H256::from_slice(&hash), ());
+
+        for id in 0..max_attestations + offset {
+            if id < offset + 1 {
+                // benchmark with double the minimum number of elements for publishing an attestation
```

Comment:
You should also fill the last one (the one that you'll publish)
```suggestion
            if id < offset + 1 || id == max_attestations + offset - 1 {
```